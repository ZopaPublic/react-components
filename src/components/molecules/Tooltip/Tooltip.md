### Summary

This is just a wrapper around [`@tippyjs/react`](https://github.com/atomiks/tippy.js-react) with some style constraints applied.

For more information:

- [Props](https://atomiks.github.io/tippyjs/all-props/)
- [Examples](https://atomiks.github.io/tippyjs/)

### Requisites 🗯

You should include `<Tooltip.Styles>` in you top-most level component, preferably like this:

```tsx static
import { Tooltip, GlobalStyles } from '@zopauk/react-components';

const App = () => (
  <>
    <Routes />
    <GlobalStyles />
    <Tooltip.Styles />
  </>
);
```

So that styles are applied once and not every time the user navigates to a particular route.

### Tips 💄

- If you want the tooltip target to be a React component, you [need to wrap it with `forwardRef`](https://github.com/atomiks/tippyjs-react#component-children).
- We don't allow to customise `theme`, `animation` and `flipOnUpdate` to make sure the tooltip UX stays the same.
- By default you won't be able to pass JSX to the tooltip content. You can change that via `allowHTML={true}` prop.

### Example

```tsx
import { forwardRef } from 'react';
import { Tooltip, Button } from '@zopauk/react-components';

function Example() {
  return (
    <>
      <Tooltip.Styles />
      <Tooltip content="I prefer 🍕 over 🍰 ...">
        <span>💁🏻‍♂️</span>
      </Tooltip>
    </>
  );
}

<Example />;
```
