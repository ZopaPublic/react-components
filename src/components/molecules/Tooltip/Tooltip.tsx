import React from 'react';
import Tippy, { TippyProps } from '@tippyjs/react';
import TooltipStyles from './TooltipStyles/TooltipStyles';

Tooltip.Styles = TooltipStyles;

export default function Tooltip({ children, ...config }: Omit<TippyProps, 'animation' | 'theme' | 'flipOnUpdate'>) {
  return (
    <Tippy allowHTML={false} placement="top" animation="shift-toward" inertia={true} {...config}>
      {children}
    </Tippy>
  );
}
