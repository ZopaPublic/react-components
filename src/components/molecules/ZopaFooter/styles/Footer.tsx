import styled from 'styled-components';

export const Footer = styled.footer`
  background-color: ${({ theme }) => theme.footer.bgColor};
`;
