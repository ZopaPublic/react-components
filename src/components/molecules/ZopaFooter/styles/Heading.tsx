import styled from 'styled-components';
import Heading from '../../../atoms/Heading/Heading';

export const FooterHeading = styled(Heading)`
  text-transform: uppercase;
`;
