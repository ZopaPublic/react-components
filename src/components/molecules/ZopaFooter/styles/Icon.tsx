import styled from 'styled-components';

export const Icon = styled.span<{ variant: string }>`
  display: inline-block;
  width: 32px;
  height: 32px;
  background: url(${({ variant }) => variant}) no-repeat;
`;
