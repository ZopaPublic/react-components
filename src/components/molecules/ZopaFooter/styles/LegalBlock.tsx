import styled from 'styled-components';
import FlexCol from '../../../layout/FlexCol/FlexCol';
import { grid } from '../../../../constants';

export const LegalBlock = styled(FlexCol)`
  order: 3;

  @media (min-width: ${grid.breakpoints.l}px) {
    order: 2;
  }
`;
