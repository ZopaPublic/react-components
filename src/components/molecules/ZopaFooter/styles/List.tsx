import styled from 'styled-components';

export const List = styled.ul`
  margin: 0;
  padding: 0;
  list-style-type: none;
`;
