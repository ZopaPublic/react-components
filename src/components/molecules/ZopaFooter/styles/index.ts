export { Footer } from './Footer';
export { FooterHeading as Heading } from './Heading';
export { Icon } from './Icon';
export { LegalBlock } from './LegalBlock';
export { List } from './List';
export { LogoBlock } from './LogoBlock';
export { SocialBlock } from './SocialBlock';
export { SocialLink } from './SocialLink';
