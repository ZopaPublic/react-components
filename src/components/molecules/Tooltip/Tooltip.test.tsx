import axe from '../../../../axe-helper';
import React from 'react';
import { render } from '@testing-library/react';
import { getGlobalStyleTags } from '../../../helpers/test/styles';
import Tooltip from './Tooltip';

describe('<Tooltip />', () => {
  it('renders the component with no a11y violations', async () => {
    const { container } = render(
      <Tooltip content="I prefer 🍕 over 🍰 ...">
        <span>💁🏻‍♂️</span>
      </Tooltip>,
    );

    expect(container.firstChild).toMatchSnapshot();
    const results = await axe(container.innerHTML);
    expect(results).toHaveNoViolations();
  });

  it('allows to set global styles', () => {
    render(<Tooltip.Styles />);

    const [globalModalStyles] = getGlobalStyleTags();
    expect(globalModalStyles).toMatchSnapshot();
  });
});
