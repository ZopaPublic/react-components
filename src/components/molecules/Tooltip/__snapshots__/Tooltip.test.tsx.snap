// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> allows to set global styles 1`] = `"/* sc-component-id: sc-global-2866731758 */"`;

exports[`<Tooltip /> renders the component with no a11y violations 1`] = `
<span>
  💁🏻‍♂️
</span>
`;
